sdk/communication/AzureCommunicationChat/Source/ChatClient.swift

```diff
@@ -330,4 +333,154 @@ public class ChatClient {
+            completionHandler(.failure(AzureError.client("Failed to initialize PushNotificationClient.")))
+            return
+        }
+
+        // After successful initialization, start push notifications
+        pushNotificationClient.startPushNotifications(
+            deviceRegistrationToken: deviceToken,
+            encryptionKeys: encryptionKeys
+        ) { result in
+            switch result {
+            case let .success(response):
+                completionHandler(.success(response))
+            case let .failure(error):
+                self.options.logger
+                    .error("Failed to start push notifications with error: \(error.localizedDescription)")
+                completionHandler(.failure(AzureError.client("Failed to start push notifications", error)))
+            }
+        }
+    }
+
+    /// Stop push notifications.
+    /// - Parameter completionHandler: Success indicates push notifications have been stopped.
+    public func stopPushNotifications(
+        completionHandler: @escaping (Result<HTTPResponse?, AzureError>) -> Void
+    ) {
+        // If PushNotification has already been stopped, return success and add a warning.
+        guard self.pushNotificationClient?.pushNotificationsStarted == true else {
+            options.logger.warning("Warning: PushNotification has already been stopped.")
+            completionHandler(.success(nil))
+            return
+        }
+
+        // Report an error if pushNotificationClient doesn't exist
+        guard let pushNotificationClient = pushNotificationClient else {
+            completionHandler(.failure(
+                AzureError
+                    .client(
+                        "PushNotificationClient is not initialized, cannot stop push notificaitons. Ensure startPushNotifications() is called first."
+                    )
+            ))
+            return
+        }
+
+        // Stop push notification
+        pushNotificationClient.stopPushNotifications { result in
+            switch result {
+            case let .success(response):
+                let refreshedRegistrationId = UUID().uuidString
+                self.registrationId = refreshedRegistrationId
+                completionHandler(.success(response))
+            case let .failure(error):
+                self.options.logger
+                    .error("Failed to stop push notifications with error: \(error.localizedDescription)")
+                completionHandler(.failure(AzureError.client("Failed to stop push notifications", error)))
+            }
+        }
+    }
+
+    /// Handle the data payload for an incoming push notification.
+    /// - Parameters:
+    ///   - notification: The APNS push notification payload ( including "aps" and "data" )
+    ///   - encryptionKeys: An array of keyPair used for verification & decryption
+    public func decryptPayload(
+        notification: [AnyHashable: Any],
+        encryptionKeys: [[String: String]]
```

Comment:
I'm not sure why this structure is different than the earlier use. In any case, the SDK should handle this. The customer should only need to communicate to the SDK where the keys should be stored. Punting the key management essentially 100% to the customer makes no sense.

Reply:
If we expose a keyGenerator to the customer, such as chatClient.generateKey(), is it ok?

Reply:
As mentioned during the call earlier this week, some customers will have strict policies on how keys are managed. As for SDK provider, we need to enable customers to perform this manually. It is certainly good to provide a convenience layer to help customers manage keys. In terms of priority, it should be a nice-to-have. We also mentioned that we will provide guidance in our documentation on key management.

Reply:
I'd recommend proactively escalating this design to the archboard. This forces complexity on Contoso that they didn't ask for, and that the SDK could very well handle internally. From a usability standpoint, I am baffled why a customer would choose us over any competitor that doesn't require them to jump through these hoops.

Reply:
Since this is simply a feature branch, we can defer this for now. When the entire feature branch is ready, that it what should go to the archboard. It doesn't make sense to write champion scenarios for a PR that is only part of a feature.

Reply:
We can submit a review to Archboard once the whole feature branch is ready.